src/libfmt_macros/lib.rs

```diff
@@ -285,19 +289,24 @@ impl<'a> Parser<'a> {
     }
 
     /// Optionally consumes the specified character. If the character is not at
-    /// the current position, then the current iterator isn't moved and false is
-    /// returned, otherwise the character is consumed and true is returned.
+    /// the current position, then the current iterator isn't moved and `false` is
+    /// returned, otherwise the character is consumed and `true` is returned.
     fn consume(&mut self, c: char) -> bool {
-        if let Some(&(_, maybe)) = self.cur.peek() {
+        self.consume_pos(c).is_some()
+    }
+
+    /// Optionally consumes the specified character. If the character is not at
+    /// the current position, then the current iterator isn't moved and `None` is
+    /// returned, otherwise the character is consumed and the current position is
+    /// returned.
+    fn consume_pos(&mut self, c: char) -> Option<usize> {
+        if let Some(&(pos, maybe)) = self.cur.peek() {
```

Comment:
Consider rewriting as:
```rust
    fn consume_pos(&mut self, c: char) -> Option<usize> {
        self.cur.peek()
            .filter(|&(pos, maybe)| c == maybe)
            .map(|(pos, _)| {
                self.cur.next();
                pos
            })
```

Reply:
Can't borrow `self.cur` twice in this case.

Reply:
Ah; use `.cloned()` + a let binding or something to make the first borrow end early then... or maybe this is not important. idk :)

Reply:
At that point I feel that the plain for loop ends up being easier on the eyes of the next person reading this code :)